hack/pin-dependency.sh

```diff
@@ -60,6 +60,23 @@ go get -d "${dep}@${sha}"
 
 # Find the resolved version
 rev=$(go mod edit -json | jq -r ".Require[] | select(.Path == \"${dep}\") | .Version")
+
+# No entry in go.mod, we must be using the natural version indirectly
+if [[ -z "${rev}" ]]; then
+  # backup the go.mod file, since go list modifies it
+  cp go.mod go.mod.tmp
```

Comment:
If line 69 errors, we'll have a leftover temp file in the tree. We should do something like https://github.com/kubernetes/kubernetes/blob/960055596933c4eea3f86c1648b1519b8cc0c216/hack/verify-vendor-licenses.sh#L33-L43, or at the very least, just copy the file into `${KUBE_ROOT}/_tmp` so that we can ensure it doesn't get committed.

Reply:
changed to use tmp dir and cleanup at the end